tests/integration_tests/check_messages.py

```diff
@@ -0,0 +1,341 @@
+
+    for detection in message.detections:
+        d = {
+            "confidence": detection.confidence,
+            "label": detection.label,
+            "x_center": detection.rotated_rect.center.x,
+            "y_center": detection.rotated_rect.center.y,
+            "width": detection.rotated_rect.size.width,
+            "height": detection.rotated_rect.size.height,
+            "angle": detection.rotated_rect.angle,
+            "keypoints": [[kp.x, kp.y, kp.confidence] for kp in detection.keypoints],
+            "mask": message.masks,
+        }
+        predicted_detections.append(d)
+
+    assert (
+        len(predicted_detections) == len(expected_detections)
+    ), f"The number of detections is different. Got {len(predicted_detections)}, expected {len(expected_detections)}"
+    print(
+        f"Expected number of detections: {len(expected_detections)}, predicted number of detections: {len(predicted_detections)}"
+    )
+
+    for i, expected_detection in enumerate(expected_detections):
+        predicted_detection = predicted_detections[i]
+        assert np.allclose(
+            expected_detection["confidence"],
+            predicted_detection["confidence"],
+            atol=0.01,
+        ), f"Expected confidence {expected_detection['confidence']}, got {predicted_detection['confidence']}."
+        assert (
+            expected_detection["label"] == predicted_detection["label"]
+        ), f"Expected label {expected_detection['label']}, got {predicted_detection['label']}."
+        assert np.allclose(
+            expected_detection["x_center"], predicted_detection["x_center"], atol=0.01
+        ), f"Expected x_center {expected_detection['x_center']}, got {predicted_detection['x_center']}."
+        assert np.allclose(
+            expected_detection["y_center"], predicted_detection["y_center"], atol=0.01
+        ), f"Expected y_center {expected_detection['y_center']}, got {predicted_detection['y_center']}."
+        assert np.allclose(
+            expected_detection["width"], predicted_detection["width"], atol=0.01
+        ), f"Expected width {expected_detection['width']}, got {predicted_detection['width']}."
+        assert np.allclose(
+            expected_detection["height"], predicted_detection["height"], atol=0.01
+        ), f"Expected height {expected_detection['height']}, got {predicted_detection['height']}."
+        assert np.allclose(
+            expected_detection["angle"], predicted_detection["angle"], atol=0.01
+        ), f"Expected angle {expected_detection['angle']}, got {predicted_detection['angle']}."
+
+        if "keypoints" in expected_detection.keys():
+            assert np.allclose(
+                expected_detection["keypoints"],
+                predicted_detection["keypoints"],
+                atol=0.01,
+            ), f"Expected keypoints {expected_detection['keypoints']}, got {predicted_detection['keypoints']}."
```

Comment:
This check for keypoints similarity is different than the one used in the `check_keypoints_msg`. Should we do a same check?

Reply:
Thanks for noticing, fixed in [13407fa](https://github.com/luxonis/depthai-nodes/pull/143/commits/13407fab7997f86ff2b3921c3c4242bae007e104).